weasyl/message.py

```diff
@@ -110,10 +111,19 @@ def select_submissions(userid, limit, backtime=None, nexttime=None):
                 pr.username,
                 ch.settings,
                 we.welcomeid,
-                0 AS subtype
+                0 AS subtype,
+                agg.tags
             FROM welcome we
                 INNER JOIN character ch ON we.targetid = ch.charid
                 INNER JOIN profile pr ON ch.userid = pr.userid
+                LEFT JOIN LATERAL (
```

Comment:
Based on my understanding of the keyword, it permits referencing already referenced columns; without it the subquery would be evaluated independently (see [LATERAL Subqueries](https://www.postgresql.org/docs/9.5/static/queries-table-expressions.html#QUERIES-LATERAL)). Based on how I am currently understanding the tables, the link would be ``character|submission id 1:N searchmap(char|submit) 1:1 searchtag``; the ID of the submission item needs to be compared against the ``targetid`` in the searchmap for the item being examined at that point in the overall query, hence the lateral which allows--in this instance--``ch.charid`` to be referenced and matched to any ``searchmapchar`` rows which have that ID in ``cmap.targetid``.

If there's an alternate method, let me know and I'll give it a shot; this method was the first that I found which worked, and hence why I selected it.

Not including ``LATERAL`` results in the following (stripping ``LATERAL`` without changing the subquery).

```
2016-07-19T01:34:51+0000 [stderr#error] ProgrammingError: (psycopg2cffi._impl.exceptions.ProgrammingError) invalid reference to FROM-clause entry for table "su"
2016-07-19T01:34:51+0000 [stderr#error] LINE 75:                     WHERE smap.targetid = su.submitid
2016-07-19T01:34:51+0000 [stderr#error]                                                    ^
2016-07-19T01:34:51+0000 [stderr#error] HINT:  There is an entry for table "su", but it cannot be referenced from this part of the query.
```